sql/expression/function/aggregation/sum.go

```diff
@@ -0,0 +1,144 @@
+package aggregation
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/davecgh/go-spew/spew"
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+	"gopkg.in/src-d/go-mysql-server.v0/sql/expression"
+)
+
+// Sum agregation returns the sum of all values in the selected column.
+// It implements the Aggregation interface.
+type Sum struct {
+	expression.UnaryExpression
+}
+
+// NewSum returns a new Sum node.
+func NewSum(e sql.Expression) *Sum {
+	return &Sum{expression.UnaryExpression{Child: e}}
+}
+
+// Resolved implements the Resolvable interface.
+func (m *Sum) Resolved() bool {
```

Comment:
We can use UnaryExpression implementation for this method: https://github.com/src-d/go-mysql-server/blob/c065ba6913fb686f7256eac96f08eb87d3b38607/sql/expression/common.go#L18